t/21-needles.t

```diff
@@ -168,4 +180,65 @@ subtest 'handling relative paths in update_needle' => sub {
     };
 };
 
+# subtest 'needledir set by jsonfile_in_temp_dir' => sub {
+#     # Prepare data
+#     my $cwd = Cwd::getcwd;
+#     my $jsonfile = path($cwd . '/t/data/openqa/tests/needle.json');
+#     $jsonfile->spew('{"area": [{"x" : 123, "y" : 456}]}');
+#     my $controller = OpenQA::WebAPI::Controller::File->new;
+#     $controller->app($t->app);
+#     $controller->param(name => 'needle.png');
+#     $controller->param(distri => $settings{DISTRI});
+#     $controller->param(jsonfile => $jsonfile);
+#
+#     # Call
+#     $controller->needle;
+#
+#     # Test
+#     # TODO: what should I call here?
+# };
+
+subtest 'controller->_determine_needles_dir_for_job' => sub {
+    my $controller = OpenQA::WebAPI::Controller::Step->new;
+    $controller->app($t->app);
+    $settings{CASEDIR} = 'https://something#fragment';
+    my $sandbox = tempdir(CLEANUP => 1);
+    my $job = $schema->resultset('Jobs')->create_from_settings(\%settings);
+    $job->update({result_dir => $sandbox});
+
+    subtest 'checkout_needles_sha = no' => sub {
+        $controller->app->config->{'scm git'} = {checkout_needles_sha => 'no'};
+        my ($needle_dirs, $needles_var) = $controller->_determine_needles_dir_for_job($job);
+        is($needle_dirs, undef, 'needles directory determined');
+        is($needles_var, undef, 'needles var determined');
+    };
+
+    subtest 'checkout_needles_sha = yes' => sub {
+        $controller->app->config->{'scm git'} = {checkout_needles_sha => 'yes'};
+        my ($needle_dirs, $needles_var) = $controller->_determine_needles_dir_for_job($job);
+        like($needle_dirs, qr|t/data/openqa/share/tests/fedora/needles$|, 'needles dirs is defined');
+        is($needles_var->value, 'https://something#fragment', 'needles var is defined');
+    };
+};
+
+subtest 'controller->_create_tmpdir_for_needles_refspec' => sub {
+    my $tempdir = setup_fullstack_temp_dir('21-needles');
+    my $testresult_dir = $tempdir . '21-needles/openqa/testresults/00000';
+    path($testresult_dir)->make_path;
+    my $vars_json = path($testresult_dir, 'vars.json');
+    my $vars_json_content = '{"NEEDLES_GIT_HASH": "0123456789abcdef0123456789abcdef01234567"}';
+    $vars_json->spew($vars_json_content);
```

Comment:
Remove those lines and check if the test still passes.

Reply:
It is, but I was leaving it here in order to use it to cover the eval section of the function

Reply:
With the change below I can remove this! what do you think?

Reply:
I did not suggest removing it because it's not necessary. The problem is that the vars.json is in the wrong place